lib/blackbeard/context.rb

```diff
@@ -53,15 +53,26 @@ def unique_identifier
     end
 
     def user_id
+      unless bbd.nil?
+        VisitorUserTracker.set_visitor_id_for_user(user_id: @user.id, visitor_id: bbd)
```

Comment:
Can we invert the `nil?` to be `presence`? Also, since we are setting the vid here, can we just use that value throughout?